src/sentences/dpt.rs

```diff
@@ -0,0 +1,135 @@
+use nom::character::complete::char;
+use nom::combinator::opt;
+use nom::number::complete::double;
+use nom::IResult;
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::Error;
+use crate::ParseResult;
+use crate::SentenceType;
+
+// DPT - Depth of Water
+//         1   2   3   4
+//         |   |   |   |
+//  $--DPT,x.x,x.x,x.x*hh<CR><LF>
+// Field Number:
+
+// 1. Water depth relative to transducer, meters
+
+// 2. Offset from transducer, meters positive means distance from transducer to water line negative means distance from transducer to keel
+
+// 3. Maximum range scale in use (NMEA 3.0 and above)
+
+// 4. Checksum
+
+// Example: $INDPT,2.3,0.0*46
+// $SDDPT,15.2,0.5*68 - $SDDPT is the sentence identifier (SD for the talker ID, DPT for Depth)
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct DptData {
+    pub water_depth: Option<f64>,
+    pub offset: Option<f64>,
+    pub max_range_scale: Option<f64>,
+}
+
+impl From<DptData> for ParseResult {
+    fn from(value: DptData) -> Self {
+        ParseResult::DPT(value)
+    }
+}
+
+pub fn parse_dpt_(sentence: crate::NmeaSentence) -> Result<DptData, crate::Error> {
```

Comment:
We should remove the end _ as it is a public API and it's best to keep it consistent